.test-infra/jenkins/LoadTestConfig.groovy

```diff
@@ -0,0 +1,436 @@
+     *    environment_config  : 'given_environment_config',
+     *    environment_type    : 'given_environment_type'
+     *  ]
+     * ]
+     * </blockquote></pre>
+     * @param templateConfig LoadTestConfig instance
+     * @param cl Closure with fields setting
+     * @return configuration map
+     * @see LoadTestConfig
+     * @see LoadTestConfig#templateConfig
+     */
+    static Map<String, Serializable> fromTemplate(final LoadTestConfig templateConfig, final Closure cl = {}) {
+        final def newConfig = of(templateConfig)
+        delegateAndInvoke(newConfig, cl)
+        final def properties = newConfig.propertiesMap
+        verifyProperties(properties)
+        return ConfigHelper.convertProperties(properties)
+    }
+
+    /**
+     * Returns configuration map (see {@link LoadTestConfig#fromTemplate}) directly from given settings
+     * @param cl Closure with settings
+     * @return configuration map
+     */
+    static Map<String, Serializable> config(final Closure cl = {}) {
+        final def config = new LoadTestConfig()
+        delegateAndInvoke(config, cl)
+        final def properties = config.propertiesMap
+        verifyProperties(properties)
+        return ConfigHelper.convertProperties(config.propertiesMap)
+    }
+
+    private static void delegateAndInvoke(final delegate, final Closure cl = {}) {
+        final def code = cl.rehydrate(delegate, this, this)
+        code.resolveStrategy = Closure.DELEGATE_ONLY
+        code()
+    }
+
+    private static LoadTestConfig of(final LoadTestConfig oldConfig) {
+        final def newConfig = new LoadTestConfig()
+
+        //primitive values
+        InvokerHelper.setProperties(newConfig, oldConfig.propertiesMap)
+
+        //non-primitive values
+        newConfig._pipelineOptions = oldConfig._pipelineOptions ? PipelineOptions.of(oldConfig._pipelineOptions) : null
+
+        return newConfig
+    }
+
+    @Override
+    Map<String, Serializable> toPrimitiveValues() {
+        final def map = propertiesMap
+        verifyProperties(map)
+        return ConfigHelper.convertProperties(map)
+    }
+
+    LinkedHashMap<String, Object> getPropertiesMap() {
+        return [
+                _title: _title,
+                _test: _test,
+                _runner: _runner,
+                _pipelineOptions: _pipelineOptions
+        ]
+    }
+
+    private static void verifyProperties(final LinkedHashMap<String, Object> map) {
+        for (entry in map.entrySet()) {
+            requireNonNull(entry.value, "Missing ${entry.key.substring(1)} in configuration")
+        }
+    }
+
+    private static class PipelineOptions implements SerializableOption<Map<String, Serializable>> {
+        private String _jobName
+        private String _project
+        private String  _publishToBigQuery
+        private String  _metricsDataset
+        private String  _metricsTable
+        private InputOptions  _inputOptions
+        private def  _iterations
```

Comment:
Not every load test requires `iterations` parameter. There are also tests which require totally different parameters. For example, combine test requires `top_count` parameter and this parameter is absent here.

Reply:
@kamilwu Thanks for input, I'll look into current scripts and search for other parameters. Or, maybe, do you happen to have list of all possible parameters?

Reply:
I'm afraid I don't have such a list.

I wonder whether we should explicitly specify all possible parameters. Every time we add a new parameter to one of the tests, we must also remember about this place.

What about having a fixed list of parameters (publish_to_big_query, metrics_dataset, metrics_table, input_options, runner's parameters, etc.) that every test require and accepting all the other parameters without validation?

Reply:
Will do, thanks !